### PR TITLE
Add converter for Arctica-M N1 RSR data.

### PR DESCRIPTION
**Pull request by simonrp84**
This adds spectral response functions for the MSU-GS/A instruments aboard the Arctica-M N1 satellite.
There are two instruments aboard the satellite, which have differing spectral response functions. Here I add them as `det-1` and `det-2`.

To achieve this, I have also updated `utils.py` to add the ability to save HDF5 files by detector - which wasn't implemented before.

 - [x] Tests passed: Passes ``pytest pyspectral``
 - [x] Passes ``flake8 pyspectral`` 
 - [x] Fully documented

### CODE REVIEW

File: pyspectral/utils.py
Comment (pnuu) [inline pyspectral/utils.py:247]: I think the new kwarg should be `detectors` for clarity. And have a short explanation in the docstring.

File: rsr_convert_scripts/msu_gsa_reader.py
Comment (pnuu) [inline rsr_convert_scripts/msu_gsa_reader.py:91]: And also remember to update the kwarg naming here.

**Comment by adybbroe** [comment]
@simonrp84 You can update `utils.py` with this url and bump the version to 1.0.18:
`https://zenodo.org/record/6026563/files/pyspectral_rsr_data.tgz`

**Comment by simonrp84** [comment]
Thanks for the comments @pnuu and @adybbroe! I think I've updated / addressed everything now. I also added `Arctica-M-N1` to the `INSTRUMENTS` dict in `utils.py`.

File: pyspectral/utils.py
Comment (adybbroe) [inline pyspectral/utils.py:74]: Should be `tgz` as file extension!

**Review by adybbroe — CHANGES_REQUESTED** [review]
Sorry @simonrp84 But could you please add some testing on your added code in `utils.py`? The coverage went down. I didn't pay attention to that previously....

**Comment by simonrp84** [comment]
> Sorry @simonrp84 But could you please add some testing on your added code in `utils.py`? The coverage went down. I didn't pay attention to that previously....
Looking at what's changed, it seems that the function I altered didn't have any tests before - so I would need to add them. This will probably not be a quick effort as I've not worked with mocking HDF5 before. Will see what I can do.

File: pyspectral/utils.py
Comment (adybbroe) [inline pyspectral/utils.py:274]: This seems untested. Could we improve that?

File: pyspectral/utils.py
Comment (adybbroe) [inline pyspectral/utils.py:277]: Also here, this is not tested it seems.

File: pyspectral/utils.py
Comment (adybbroe) [inline pyspectral/utils.py:291]: Same here, is this tested?

File: pyspectral/utils.py
Comment (adybbroe) [inline pyspectral/utils.py:299]: Again, not covered by tests I think.

**Review by adybbroe — COMMENTED** [review]
There are changes to utils.py which seems poorly covered by tests. Is it something you can try improve? Or do you need help?

**Comment by adybbroe** [comment]
@simonrp84 I will try see if I can help our here. Maybe as Arctica-M is kind of on hold as we have no real-time flow of data this PR is not urgent and we can see if we find some time improving the test coverage of the `convert2hdf5` function before merging. Ok?

**Comment by simonrp84** [comment]
To be honest I've forgotten what I did in this PR! Will check through it next week and then we can figure out how to move on it. Like you say, no rush now.

**Comment by simonrp84** [comment]
I have now added some tests that should improve the coverage of the code previously missed. I chose to not mock the `h5py` library but to actually reader an HDF5 RSR file through the tests and then delete it after. Am unsure if this is the best way to do things so please let me know what you think!

**Review by adybbroe — APPROVED** [review]
LGTM

**Pull request merged**
